File: ios-app/MysplitwiseCore/Sources/MysplitwiseCore/Models/Group.swift
import Foundation

public enum GroupType: String, Codable, CaseIterable {
    case trip, home, couple, other
}

/// Port of the `Group` interface in `src/lib/types.ts`.
public struct Group: Codable, Identifiable, Equatable, Hashable {
    public var id: String
    public var name: String
    public var type: GroupType
    public var memberIds: [String]
    public var simplifyDebts: Bool
    public var monthlyBudget: Double?
    public var createdAt: String

    public init(
        id: String, name: String, type: GroupType, memberIds: [String],
        simplifyDebts: Bool, monthlyBudget: Double? = nil, createdAt: String
    ) {
        self.id = id
        self.name = name
        self.type = type
        self.memberIds = memberIds
        self.simplifyDebts = simplifyDebts
        self.monthlyBudget = monthlyBudget
        self.createdAt = createdAt
    }
}

File: android-app/app/src/main/kotlin/com/mysplitwise/app/model/Group.kt
package com.mysplitwise.app.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
enum class GroupType {
    @SerialName("trip") TRIP,
    @SerialName("home") HOME,
    @SerialName("couple") COUPLE,
    @SerialName("other") OTHER,
}

/** Port of the `Group` interface in `src/lib/types.ts`. */
@Serializable
data class Group(
    val id: String,
    val name: String,
    val type: GroupType,
    val memberIds: List<String>,
    val simplifyDebts: Boolean,
    val monthlyBudget: Double? = null,
    val createdAt: String,
)
